### PR TITLE
#84 Currency converter  | Added working currency converter to integrate into MyFinances

### LINKED ISSUE #84: [IDEA] Currency converter
A page that allows the user to type in amounts, and select a currency and see the new amount. 

Also in the future maybe an option to preload the amounts from their balance, history, or graphs.

### PR DESCRIPTION
**Pull request by matthewjuarez1**
---
title:#84 Currency converter  | Added working currency converter to integrate into MyFinances
---

# Checklist

- [X] Ran the [Black Formatter]() on any new code (checks will fail without)
- [X ] Made any changes or additions to the documentation where required
- [X] Changes generate no new warnings/errors
- [X] New and existing unit tests pass locally with my changes

## What type of PR is this?

- ✨ Feature
- 📝 Documentation Update

## Description

The Currency Converter was an idea put out on the issue board. I worked on this issue and created a function, convert_currency(), that will convert a currency as needed. 
The usage for this function is documented alongside it in the code:

_convert_currency(init_currency, target_currency, amount, date=None)
Converts one currency to another, given an amount, using the forex_python library

    Parameters
    ----------
    init_currency : str, required
        The code for the initial currency to be converted

    target_currency : str, required
        The code for the target currency to be converted to

    amount : int or float, required
        The amount to be converted

    date: datetime, optional
        Past date at which the currency should be converted

    Returns
    ----------
    Returns an int or float representing the new amount_

As stated, this function will take the codes from two currencies (which are defined in UserSettings.CURRENCIES as well), an amount, and an optional date. The function will then convert the given amount from the first currency to the second, using the most current exchange rates, and return it. If a date is given, it will instead use that day's exchange rates and return the amount.  The date parameter needs to be a datetime variable, which is the same kind of variable used in the Django models. 
A ValueError will be raised if any of the 4 given parameters are not the correct datatype. 
Side Note: Forex does not have exchange rates for weekends, so if the given date is a weekend, the function will get the rates for the Friday before. 
Needed dependencies: forex_python is the main new library that this function will depend on to work properly. 
On the frontend, the currency converter can be demonstrated at /dashboard/currency_converter

## Added/updated tests?
- 🙅 no, because they aren't needed

## Related PRs, Issues etc
- Issue #84 
- Closes #84

### CODE REVIEW

**Comment by TreyWW** [comment]
Hi there @matthewjuarez1,

Thank you so much for this PR, I really appreciate your contribution. As we can see from the [test failure (line 82)](https://github.com/TreyWW/MyFinances/actions/runs/7087516394/job/19287952601?pr=138#step:7:82) `forex_currency` should be added to requirements.

Could that be added to `requirements.txt` and `requiremements.gh_actions.txt`
like:
```txt
# currency converter
forex_python==1.8
```

Apart from that, it would be nice if the print statements were changed to `message`s, so the user can see what's going on rather than the console. But I can always change that at a later date, it's no big issue.

Thanks again!

**Comment by TreyWW** [comment]
Hey @matthewjuarez1,
Any chance you could get [this change](https://github.com/TreyWW/MyFinances/pull/138#pullrequestreview-1762456234) added?

Thanks

**Comment by matthewjuarez1** [comment]
@TreyWW I can get that change in sometime this week

**Comment by matthewjuarez1** [comment]
Changes have been made, awaiting re-review

**Review by TreyWW — APPROVED** [review]

File: backend/views/core/currency_converter/dashboard.py [hunk 1]
```diff
@@ -29,11 +37,58 @@ def convert_currency(init_currency, target_currency, amount):
     if not isinstance(amount, (int, float)):
         raise ValueError("Amount is not an accepted datatype")
```
Comment (NEXUSROOKJACKSON) [inline backend/views/core/currency_converter/dashboard.py:38]: [Need data protection of currency and to stay one language and currency help](url)

Comment (TreyWW) [reply backend/views/core/currency_converter/dashboard.py:38]: ?

**Pull request merged**